server/tests/batch_comparison/test_sample_extra_batches_by_counting_group.py

```diff
@@ -421,13 +421,13 @@ def test_sample_extra_batches_min_percentage_of_jurisdiction_ballots_selected(
                     b"Container,Batch Name,Number of Ballots\n"
                     b"Absentee by Mail,Batch 1,500\n"  # HMPB group
                     b"Election Day,Batch 2,500\n"  # BMD group
-                    b"Container 3,Batch 3,500\n"
-                    b"Container 4,Batch 4,500\n"
-                    b"Container 5,Batch 5,100\n"
-                    b"Container 6,Batch 6,100\n"
-                    b"Container 7,Batch 7,100\n"
-                    b"Container 8,Batch 8,100\n"
-                    b"Container 9,Batch 9,1000000\n"  # Must be selected to hit the 2% selection threshold
+                    b"Election Day,Batch 3,500\n"
+                    b"Election Day,Batch 4,500\n"
+                    b"Election Day,Batch 5,100\n"
+                    b"Election Day,Batch 6,100\n"
+                    b"Election Day,Batch 7,100\n"
+                    b"Election Day,Batch 8,100\n"
+                    b"Election Day,Batch 9,1000000\n"  # Must be selected to hit the 2% selection threshold
```

Comment:
This test actually tests 2 things:
1. At least one HMPB and one BMD batch (as defined by the conditions in https://github.com/votingworks/arlo/pull/2011/files#diff-0ebaf00d54e8db50181143402050c27dbefe0459e118cf8f0f137b1db4fbb51bR624) are selected
2. At least 2% of the total ballots are selected

`Container` validation is turned on for this test so `Container N` is no longer a valid value here. This has the side effect of making Batch 9 a BMD group. Since Batch 9 must be selected to satisfy requirement 2 it will always be picked. Therefore the BMD group requirement is also automatically satisfied; this test no longer exhaustively tests requirement 1.

To fix this I adjusted expectations for this test and added a separate test for requirement 1 below.

Reply:
Thanks for catching this and adding an extra test!